src/main/java/com/practice/slideshow/service/KafkaLoggingService.java

```diff
@@ -22,13 +32,29 @@ public class KafkaLoggingService {
    */
   public void logAction(String actionType, String message) {
     if (actionType == null || message == null) {
+      log.warn("Action type or message is null. Skipping log action.");
       return;
     }
     String logMessage = String.format("Action: %s | Details: %s", actionType, message);
-    try {
-      kafkaTemplate.send(TOPIC, logMessage);
-    } catch (Exception e) {
-      System.err.println("Kafka send failed: " + e.getMessage());
-    }
+    String mediaType = actionType.contains("IMAGE") ? "image" : "audio";
+    Message<String> msg = MessageBuilder.withPayload(logMessage)
```

Comment:
Use producer record
You may use this record as an example https://github.com/dilipsundarraj1/kafka-for-developers-using-spring-boot/blob/main/library-events-producer/src/main/java/com/learnkafka/producer/LibraryEventProducer.java